test/object_store/test_basic.py

```diff
@@ -220,3 +224,56 @@ def list_bucket(s3_server):
         for o in objects:
             for ent in sstable_entries:
                 assert not o.startswith(str(ent[1])), f'Sstable object not cleaned, found {o}'
+
+@pytest.mark.asyncio
+async def test_memtable_flush_retries(test_tempdir, s3_server, ssl):
+    '''verify ownership table is garbage-collected on boot'''
+    ks = 'test_ks'
+    cf = 'test_cf'
+    rows = [('0', 'zero'),
+            ('1', 'one'),
+            ('2', 'two')]
+
+    print('Spoof the object-store config')
+    local_config = pathlib.Path(test_tempdir) / 'object_storage.yaml'
+    MinioServer.create_conf_file(s3_server.address, s3_server.port, 'bad_key', 'bad_secret', 'bad_region', local_config)
```

Comment:
the test is flattened into a single piece. i'd suggest extract the spoof part into a dedicated function. and split the steps with empty lines for better readability.

```py
@contextmanager
def bad_object_store_config(tempdir, s3_server, cluster):
    with open(s3_server.config_file) as f:
        orig_config = f.read()
    MinioServer.create_conf_file(s3_server.address, s3_server.port, 'bad_key', 'bad_secret', 'bad_region', s3_server.config_file)
    cluster.reload_config()
    try:
        yield
    finally:
        with open(s3_server.config_file, 'w') as f:
            f.write(orig_config)
        cluster.reload_config()
```

```py
with managed_cluster(test_tempdir, ssl, s3_server) as cluster:
    with bad_object_store_config(test_tempdir, s3_server, cluster):
        # write to it
        # flush
    # wait for flush
    # read from it
```